### DISCUSSION TITLE
Testnet/Mainnet switch

### DISCUSSION THREAD

**Discussion by bonomat** [get10101/itchysats#2927] [Ideas]
In theory it should be possible to switch between `mainnet` and `testnet` during runtime. What do you think: 
- how hard is this?
- would it be worth to give a try?

**Comment by klochowicz** [comment]
what would be the use case?

I can see the appeal of indicating somewhere what network you're running on.

**Reply by bonomat** [reply]
To allow users to play around with testnet coins and then easily do the switch and play with real coins

**Reply by holzeis** [reply]
I like that..!

**Reply by holzeis** [reply]
But do you intend to also connect to the testnet maker? in terms of version? I would like to see the same (stable) maker version for testnet as well to provide some form of stability - although a big disclaimer should be added.

**Reply by klochowicz** [reply]
IMHO it's not worth the effort as it's an UX challenge. Literally no-one will go back to testnet from mainnet, and the potential for confusion is high. Such switch should be disabled on all the full node distros as well :/

Perhaps we could just start with sane defaults? like default to testnet in the binary, whilst being clear that this is testnet? Alternatively a symlink that starts it up this way. Note that itchysats-docker already defaults to testnet.

**Reply by holzeis** [reply]
I it possible for an umbrel user to install the testnet version?

**Reply by bonomat** [reply]
> I it possible for an umbrel user to install the testnet version?

theoretically yes, practically you need to know your way around the cli.

**Reply by klochowicz** [reply]
we can easily distribute 2 binaries with different defaults: `itchysats-taker-testnet` and `itchysats-taker-mainnet` (then it can be just double-click still). Or using a symlink that encompasses the CLI option.

**Reply by klochowicz** [reply]
one thing that would make this feature suggestion simpler - if we only asked the user in the UI at startup! and, depending on the answer, launch daemon with different parameters.

**Reply by holzeis** [reply]
> one thing that would make this feature suggestion simpler - if we only asked the user in the UI at startup! and, depending on the answer, launch daemon with different parameters.

I guess with UI you are referring to the umbrel UI? in that case the solution would be very much platform dependent.

**Reply by holzeis** [reply]
IMHO, we should do everything we can to increase the trust by the user towards the system. At the end they are trusting the system with their money. Giving them a playground in one way or another to understand how the system works is in my opinion key to increase the user base - but I am not saying this has to be testnet. It could also be a video showcasing and explaining examples, etc.

However, this might not be necessary as @bonomat argued that crypto users are far more forgiving when it comes to losing their money.
